cloudferrylib/utils/drivers/ssh_chunks.py

```diff
@@ -51,128 +53,129 @@ def transfer(self, data):
-                    # If not equal - retry with 'attempts_count' times
-                    if md5_src == md5_dst:
-                        success = 1
-
-                    if not success:
-                        attempt += 1
-                        LOG.critical("Unable to transfer part %s of %s. "
-                                     "Retrying... Attempt %s from %s.",
-                                     part, path_src, attempt, attempts_count)
-                        if attempt == attempts_count:
-                            LOG.error("SSH chunks transfer of %s failed.",
-                                      path_src)
-                            break
-                            # TODO: save state:
-                            # chunks count, volume info, and all metadata info
-                            # with timestamp and reason, errors info, error
-                            # codes
-                        continue
-
-                    #  Write chunk on destination
-                    if part == range(part_count)[0]:
-                        command = dd_dst_command % (dst_temp_dir,
-                                                    part,
-                                                    path_dst,
-                                                    0,
-                                                    part_size)
-                    elif part == range(part_count)[-1] and part_modulo:
-                        command = dd_dst_command % (dst_temp_dir,
-                                                    part,
-                                                    path_dst,
-                                                    part * part_size,
-                                                    part_modulo)
-                    else:
-                        command = dd_dst_command % (dst_temp_dir,
-                                                    part,
-                                                    path_dst,
-                                                    part * part_size,
-                                                    part_size)
-                    with hide('running'):
-                        # Because of password
-                        run(ssh_command %
-                            (ssh_user_dst, host_dst,
-                             'echo %s | sudo -S %s' % (ssh_sudo_pass_dst,
-                                                       command)))
-
-                        LOG.info(
-                            'Running: %s', ssh_command %
-                            (ssh_user_dst, host_dst,
-                             'echo %s | sudo -S %s' % ('<password>', command)))
-
-                    # Delete used chunk from both servers
-                    run(rm_command % (src_temp_dir, part))
-                    run(ssh_command % (ssh_user_dst, host_dst, rm_command %
-                                       (dst_temp_dir, part)))
+        with files.RemoteDir(runner_src, temp_dir_name):
```

Comment:
I would suggest moving remote temporary dir creation to a separate class which incorporates `mkdir -u` logic with interface similar to `RemoteDir`. Also, `mkdir -u` must be run on destination node, not on the execution node, since `$TEMP` environment variable may be different there.